dspace-server-webapp/src/test/java/org/dspace/app/rest/RelationshipRestRepositoryIT.java

```diff
@@ -2482,5 +2500,160 @@ public void putRelationshipWithJson() throws Exception {
 
     }
 
+    @Test
+    public void orgUnitAndOrgUnitRelationshipVirtualMetadataTest() throws Exception {
+        context.turnOffAuthorisationSystem();
+        EntityType orgUnit = entityTypeService.findByEntityType(context, "OrgUnit");
+        RelationshipType isParentOrgUnitOf = relationshipTypeService
+            .findbyTypesAndTypeName(context, orgUnit, orgUnit, "isParentOrgUnitOf", "isChildOrgUnitOf");
+
+        MetadataSchema metadataSchema = metadataSchemaService.find(context, "relation");
+        MetadataFieldBuilder.createMetadataField(context, metadataSchema, "isParentOrgUnitOf", null, null).build();
+        MetadataFieldBuilder.createMetadataField(context, metadataSchema, "isChildOrgUnitOf", null, null).build();
+
+        String adminToken = getAuthToken(admin.getEmail(), password);
+
+        context.restoreAuthSystemState();
+        // Here we create our first Relationship to the Publication to give it a dc.contributor.author virtual
+        // metadata field.
+        MvcResult mvcResult = getClient(adminToken).perform(post("/api/core/relationships")
```

Comment:
This `post()` and all the others in this IT do not seem to be cleaning up after themselves.  Remember we should be using a try/finally, using an `.andDo()` to capture the ID of the created object (I suspect this code was just written prior to that best practice being established).  In any case, here's an example: https://github.com/DSpace/DSpace/blob/main/dspace-server-webapp/src/test/java/org/dspace/app/rest/ScriptRestRepositoryIT.java#L195-L207

Reply:
This has been updated